Sources/KSCrashRecording/Monitors/KSCrashMonitor_Memory.m

```diff
@@ -95,22 +128,39 @@ + (instancetype)new NS_UNAVAILABLE;
 
 @property (atomic, readonly) KSCrash_ApplicationTransitionState transitionState;
 
+- (void)addObserver:(id<AppStateTrackerObserving>)observer;
+- (id<AppStateTrackerObserving>)addObserverWithBlock:(AppStateTrackerBlockObserverBlock)block;
+
+- (void)removeObserver:(id<AppStateTrackerObserving>)observer;
+
 @end
 
 @implementation AppStateTracker
 
 + (void)load
 {
     g_AppStateTracker = [[AppStateTracker alloc] initWithNotificationCenter:NSNotificationCenter.defaultCenter];
+    [g_AppStateTracker addObserver:self];
     [g_AppStateTracker start];
 }
 
++ (void)appStateTrackerDidChangeApplicationTransitionState:(KSCrash_ApplicationTransitionState)transitionState
+{
+    _ks_memory_update(^(KSCrash_Memory *mem) {
+        mem->state = transitionState;
+    });
+}
+
 - (instancetype)initWithNotificationCenter:(NSNotificationCenter *)notificationCenter
 {
     if (self = [super init]) {
+        _lock = OS_UNFAIR_LOCK_INIT;
+        _observers = [NSMutableArray array];
```

Comment:
Maybe for observers it's better to use `NSPointerArray` with `weak` pointers to prevent memory leaks?

Reply:
You have a point. However, I was trying to be intentional about lifetime by forcing a remove if that is ever required, which would likely be rare for a class like this. NSPointerArray lacks a defined lifetime for weak objects it holds other than they'll be nilled sometime in the future. Is that acceptable to you?

Reply:
I won't use an NSPointerArray since we're dealing with two types of objects (blocks or an object that implements the protocol), but I'll make use of `AppStateTrackerBlockObserver` and it'll contain a weak reference to an observer if one was passed in which will have the same effect.